command/repo.go

```diff
@@ -0,0 +1,57 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	RootCmd.AddCommand(repoCmd)
+	repoCmd.AddCommand(repoViewCmd)
+}
+
+var repoCmd = &cobra.Command{
+	Use:   "repo",
+	Short: "View repositories",
+	Long: `Work with GitHub repositories.
+
+A repository can be supplied as an argument in any of the following formats:
+- by owner/repo, e.g. "cli/cli"
+- by URL, e.g. "https://github.com/cli/cli"`,
+}
+
+var repoViewCmd = &cobra.Command{
+	Use:   "view [{<owner/repo> | <url>}]",
+	Short: "View a repository in the browser",
+	Long: `View a repository specified by the argument in the browser.
+
+Without an argument, the repository that belongs to the current
+branch is opened.`,
+	RunE: repoView,
+}
+
+func repoView(cmd *cobra.Command, args []string) error {
+	ctx := contextForCommand(cmd)
+	baseRepo, err := ctx.BaseRepo()
```

Comment:
I think for consistency this should use the new `determineBaseRepo` helper

Reply:
Sounds great, updated.